sale_open_qty/models/sale_order.py

```diff
@@ -54,7 +54,7 @@ def _search_qty_to_deliver(self, operator, value):
             [('qty_to_deliver', operator, value)])
         order_ids = so_lines.mapped('order_id')
         res.append(('id', 'in', order_ids.ids))
-        return res
+        return [('order_line.qty_to_deliver', operator, value)]
```

Comment:
@andreampiovesana And remove all code above... :smile:

Reply:
do you mean this?

    @api.model
    def _search_qty_to_deliver(self, operator, value):
        return [('order_line.qty_to_deliver', operator, value)]

Reply:
in purchase_open_qty :

    @api.model
    def _search_pending_qty_to_receive(self, operator, value):
        if operator != '=' or not isinstance(value, bool):
            raise ValueError(_("Unsupported search operator"))
        po_line_obj = self.env['purchase.order.line']
        po_lines = po_line_obj.search(
            [('qty_to_receive', '>', 0.0)])
        orders = po_lines.mapped('order_id')
        if value:
            return [('id', 'in', orders.ids)]
        else:
            return [('id', 'not in', orders.ids)]

Reply:
Yes. As qty_to_deliver is stored, you can search on it.

What does the search function do eventually? It lauches a search on Sale Order with that domain.

So, eventually, it will call self.env['sale.order'].search([('order_line.qty_to_deliver', operator, value)]).

If we can avoid :

- A search on order_lines
- A search on order ids

=> two searches

Reply:
You can make a PR on purchase_open_qty to optimize that :smiley:

Reply:
@andreampiovesana Could you remove the unused code ? Thanks